Sources/BraintreeCore/Configuration/ConfigurationLoader.swift

```diff
@@ -37,16 +37,16 @@ class ConfigurationLoader {
     ///   - `Error?`: An error object if fetching the configuration fails or if the instance is deallocated.
     @_documentation(visibility: private)
     func getConfig() async throws -> BTConfiguration {
-        if let existingTask {
-            return try await existingTask.value
-        }
-
         if let cachedConfig = try? configurationCache.getFromCache(authorization: http.authorization.bearer) {
             return cachedConfig
         }
+        
+        if let existingTask {
+            return try await existingTask.value
```

Comment:
I had put existingTask check first because I wanted to avoid cache simultaneous read/write scenario, however rare.